FrontEnd/src/Appointment/AppointmentPage.jsx

```diff
@@ -0,0 +1,119 @@
+import { useState, useEffect } from "react";
+import {
+  fetchPendingAppointments,
+  fetchUpcomingAppointments,
+  acceptAppointment,
+  declineAppointment,
+} from "../BookingPage/BookingDB";
+import "./AppointmentPage.css";
+import AppointmentDetails from "./AppointmentDetails";
+import Modal from "../Modal";
+
+function AppointmentPage({ userData }) {
+  const [isAppointmentDetailsModalShown, setIsAppointmentDetailsModalShown] =
+    useState(false);
+  const [pendingAppointmentData, setPendingAppointmentData] = useState([]);
+  const [upcomingAppointmentData, setUpcomingAppointmentData] = useState([]);
+  const userID = userData?.userID;
+
+  useEffect(() => {
+    if (userID === undefined) {
+      return;
+    }
+    const unsubscribe = fetchPendingAppointments(userID, (appointmentData) => {
+      setPendingAppointmentData(appointmentData);
+    });
+
+    return () => unsubscribe();
+  }, [userID]);
+
+  useEffect(() => {
+    if (userID === undefined) {
+      return;
+    }
+    const unsubscribe = fetchUpcomingAppointments(userID, (appointmentData) => {
+      setUpcomingAppointmentData(appointmentData);
+    });
+
+    return () => unsubscribe();
+  }, [userID]);
+
+  const acceptedAppointment = async (userID, vendorID, appointmentID) => {
```

Comment:
`acceptedAppointment` sounds like a variable, not a function

```suggestion
  const handleAcceptAppointment = async (userID, vendorID, appointmentID) => {
```